### PR TITLE
Split dashboard header buttons into files

### PR DESCRIPTION
**Pull request by oisincoveney**
Splits the buttons in the Dashboard header from a mess of a ton of components into different button components in a bunch of new files. The functionality should not have changed (so the verification should be that the tests will pass). 

The idea behind this PR is that I'll split the buttons into different components, then in a separate PR to set up a system for adding/removing/changing the order of buttons, as we'll want this functionality for the embedding SDK.

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/DashboardHeader/buttons/SaveEditButton.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  setEditingDashboard,
+  updateDashboardAndCards,
+} from "metabase/dashboard/actions";
+import { getMissingRequiredParameters } from "metabase/dashboard/selectors";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { dismissAllUndo } from "metabase/redux/undo";
+import { Tooltip } from "metabase/ui";
+import type { UiParameter } from "metabase-lib/v1/parameters/types";
+
+export const SaveEditButton = (props: { actionFn: () => void }) => {
```
Comment (heypoom) [inline frontend/src/metabase/dashboard/components/DashboardHeader/buttons/SaveEditButton.tsx:16]: should we rename `actionFn` here to `onDoneEditing`?

**Review by heypoom — COMMENTED** [review]
Looks great! Just a few observations on the `key` prop's placements, and other minor things 🙏🏻

**Review by deniskaber — APPROVED** [review]
Agree that we should remove extra `key=`s.
Also I think if a piece of logic is used in a single place and it's not huge, we should not move it into a hook. For me it becomes more difficult to read, as you need to navigate to hook content, and it switches focus from component context.

Apart from that, awesome effort! Thanks!

File: frontend/src/metabase/dashboard/components/DashboardHeader/buttons/AddActionElementButton.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          action: {},
+        }),
+      );
+    }
+  };
+
+  return (
+    <>
+      <DashboardHeaderActionDivider />
+      <Tooltip key="add-action-button" label={t`Add action button`}>
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/dashboard/components/DashboardHeader/buttons/AddActionElementButton.tsx:33]: Yeah I think a bunch of these were just me forgetting to actually rmove the keys after doing all of the refactoring

File: frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.tsx [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
-}) => {
+export const DashboardEmbedAction = () => {
+  const dashboard = useSelector(getDashboardComplete);
+
   const [isModalOpen, setIsModalOpen] = useState(false);
 
-  return (
+  return dashboard ? (
     <>
       <EmbedMenu
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.tsx:16]: you sure? We have `DashboardSharingEmbeddingModal` there as well

File: frontend/src/metabase/dashboard/components/DashboardHeader/buttons/SaveEditButton.tsx (hunk 1 above)
Comment (oisincoveney) [reply frontend/src/metabase/dashboard/components/DashboardHeader/buttons/SaveEditButton.tsx:16]: yes!

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -252,13 +238,10 @@ function Dashboard(props: DashboardProps) {
   );
 
   const handleAddQuestion = useCallback(() => {
-    handleSetEditing(dashboard);
+    onRefreshPeriodChange(null);
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx:241]: Oh nice, if that's intentional then we should be good to go.

File: frontend/src/metabase/dashboard/components/DashboardHeader/buttons/AddActionElementButton.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+          displayType: "button",
+          action: {},
+        }),
+      );
+    }
+  };
+
+  return (
+    <>
+      <DashboardHeaderActionDivider />
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/dashboard/components/DashboardHeader/buttons/AddActionElementButton.tsx:32]: moved it into its own thing in `DashboardHeader`

File: frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.tsx:16]: @oisincoveney oh, the GitHub diffs fooled my eyes, I didn't noticed that there is more content that was collapsed. you can ignore this 🙏🏻

**Pull request merged**